### PR TITLE
WIP: Packages: Add the code editor package

### PR DESCRIPTION
**Pull request by youknowriad**
This PR adds the code editor package as a Gutenberg package. The code was copied as is from Core and tweaked a little bit:

 - Removing the use of globals
 - Lodash instead of undersscore

This PR is not working yet because in the CodeEditor component, we try to do async loading of scripts and styles and for some reason the `load-scripts.php` is not returning the `wp-code-editor` script and style (registered in Gutenberg), it only works for Core registered scripts/styles (I'll need a help from a WordPress Core expert here).

### CODE REVIEW

**Comment by aduth** [comment]
Related: #8017

File: package.json [hunk 1]
```diff
@@ -21,9 +21,11 @@
 		"@wordpress/blob": "file:packages/blob",
 		"@wordpress/block-serialization-spec-parser": "file:packages/block-serialization-spec-parser",
 		"@wordpress/blocks": "file:packages/blocks",
+		"@wordpress/code-editor": "file:packages/code-editor",
 		"@wordpress/components": "file:packages/components",
 		"@wordpress/compose": "file:packages/compose",
 		"@wordpress/core-data": "file:packages/core-data",
+		"@wordpress/core-editor": "file:packages/code-editor",
```
Comment (aduth) [inline package.json]: Is this line intentional? `core-editor` , not `code-editor` ?

File: packages/code-editor/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Blob
```
Comment (aduth) [inline packages/code-editor/README.md]: Copypasta. 🍝

File: packages/code-editor/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+Code Editor for WordPress.
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/code-editor --save
```
Comment (aduth) [inline packages/code-editor/README.md]: `--save` is not necessary (it's now default).

File: packages/code-editor/src/index.js [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+ * @param {Function}   settings.onChangeLintingErrors - Callback for when there are changes to linting errors.
+ * @param {Function}   settings.onUpdateErrorNotice - Callback to update error notice.
+ *
+ * @return {void}
+ */
+function configureLinting( editor, settings ) {
+	let currentErrorAnnotations = [],
+		previouslyShownErrorAnnotations = [];
+
+		/**
```
Comment (aduth) [inline packages/code-editor/src/index.js]: Tabbing alignment (various related issues with most functions in this file).

File: package.json (hunk 1 above)
Comment (youknowriad) [reply package.json]: definitely not :)

File: packages/code-editor/src/index.js (hunk 4 above)
Comment (youknowriad) [reply packages/code-editor/src/index.js]: You know, old style WP scripts 🤷‍♂️ I did some tweaks, but there more we can do here :)

**Comment by youknowriad** [comment]
Blocked by a reliable lazy-loading approach

**Comment by youknowriad** [comment]
Not anymore. We may want something like that later but fine to close now.

**Pull request closed without merging**